internal/service/dynamodb/global_secondary_index_test.go

```diff
@@ -0,0 +1,1139 @@
+				),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue(resourceNameTable, tfjsonpath.New("ttl"), knownvalue.ListExact([]knownvalue.Check{
+						knownvalue.ObjectExact(map[string]knownvalue.Check{
+							"attribute_name":  knownvalue.StringExact(""),
+							names.AttrEnabled: knownvalue.Bool(false),
+						}),
+					})),
+				},
+			},
+			{
+				ResourceName:      resourceNameGSI,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccGlobalSecondaryIndexConfig_billingPayPerRequest_onDemandThroughputWithCapacity(rNameTable, rNameGSI, 2),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckInitialTableExists(ctx, resourceNameTable, &conf),
+					testAccCheckInitialGSIExists(resourceNameGSI, &conf, &gsi),
+
+					resource.TestCheckResourceAttr(resourceNameGSI, names.AttrName, rNameGSI),
+					resource.TestCheckResourceAttr(resourceNameGSI, "table", rNameTable),
+					resource.TestCheckResourceAttr(resourceNameGSI, "hash_key", rNameTable),
+					resource.TestCheckResourceAttr(resourceNameGSI, "hash_key_type", "S"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "range_key", rNameGSI),
+					resource.TestCheckResourceAttr(resourceNameGSI, "range_key_type", "S"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "projection_type", "ALL"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "read_capacity", "0"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "write_capacity", "0"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "on_demand_throughput.#", "1"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "on_demand_throughput.0.%", "2"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "on_demand_throughput.0.max_read_request_units", "2"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "on_demand_throughput.0.max_write_request_units", "2"),
+				),
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue(resourceNameTable, tfjsonpath.New("ttl"), knownvalue.ListExact([]knownvalue.Check{
+						knownvalue.ObjectExact(map[string]knownvalue.Check{
+							"attribute_name":  knownvalue.StringExact(""),
+							names.AttrEnabled: knownvalue.Bool(false),
+						}),
+					})),
+				},
+			},
+			{
+				ResourceName:      resourceNameGSI,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckInitialGSIExists(n string, tbl *awstypes.TableDescription, gsi *awstypes.GlobalSecondaryIndexDescription) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+
+		if tbl == nil {
+			return errors.New("table config is empty")
+		}
+
+		for _, g := range tbl.GlobalSecondaryIndexes {
+			if rs.Primary.Attributes[names.AttrName] == aws.ToString(g.IndexName) {
+				*gsi = g
+
+				break
+			}
+		}
```

Comment:
Look up the GSI using `findGSIByTwoPartKey`. You will likely have to export it by adding it to `exports_test.go`.

This function will then no longer need the parameter `tbl`